compute/cloud-client/src/main/java/compute/reservation/CreateInstanceNotConsumeReservation.java

```diff
@@ -0,0 +1,122 @@
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package compute.reservation;
+
+// [START compute_instance_not_consume_reservation]
+import static com.google.cloud.compute.v1.ReservationAffinity.ConsumeReservationType.NO_RESERVATION;
+
+import com.google.api.gax.longrunning.OperationFuture;
+import com.google.cloud.compute.v1.AttachedDisk;
+import com.google.cloud.compute.v1.AttachedDiskInitializeParams;
+import com.google.cloud.compute.v1.InsertInstanceRequest;
+import com.google.cloud.compute.v1.Instance;
+import com.google.cloud.compute.v1.InstancesClient;
+import com.google.cloud.compute.v1.NetworkInterface;
+import com.google.cloud.compute.v1.Operation;
+import com.google.cloud.compute.v1.ReservationAffinity;
+import java.io.IOException;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+public class CreateInstanceNotConsumeReservation {
+  public static void main(String[] args)
+      throws IOException, ExecutionException, InterruptedException, TimeoutException {
+    // TODO(developer): Replace these variables before running the sample.
+    // Project ID or project number of the Cloud project you want to use.
+    String projectId = "YOUR_PROJECT_ID";
+    // Name of the zone you want to use.
+    String zone = "us-central1-a";
+    // Name of the VM instance you want to query.
+    String instanceName = "YOUR_INSTANCE_NAME";
+
+    createInstanceNotConsumeReservation(projectId, zone, instanceName);
+  }
+
+  // Create a virtual machine that explicitly doesn't consume reservations
+  public static Instance createInstanceNotConsumeReservation(
+      String project, String zone, String instanceName)
+      throws IOException, InterruptedException, ExecutionException, TimeoutException {
+    // Below are sample values that can be replaced.
+    // machineType: machine type of the VM being created.
+    // *   This value uses the format zones/{zone}/machineTypes/{type_name}.
+    // *   For a list of machine types, see https://cloud.google.com/compute/docs/machine-types
+    // sourceImage: path to the operating system image to mount.
+    // *   For details about images you can mount, see https://cloud.google.com/compute/docs/images
+    // diskSizeGb: storage size of the boot disk to attach to the instance.
+    // networkName: network interface to associate with the instance.
+    String machineType = String.format("zones/%s/machineTypes/n1-standard-1", zone);
+    String sourceImage = "projects/debian-cloud/global/images/family/debian-11";
+    long diskSizeGb = 10L;
+    String networkName = "default";
```

Comment:
nit: consider using `final` to emphasize that these values are used as fixed values in this method.